apache-rat-core/src/main/java/org/apache/rat/Documentation.java

```diff
@@ -0,0 +1,152 @@
+        for (ILicense l : cfg.getLicenses(LicenseFilter.all)) {
+            printLicense(l, writer);
+        }
+
+        writer.write(String.format("%n>>>> MATCHERS (Datatype IHeaderMatcher) <<<<%n%n"));
+        for (Class<? extends AbstractBuilder> mClazz : MatcherBuilderTracker.INSTANCE.getClasses()) {
+            try {
+                AbstractBuilder builder = mClazz.getConstructor().newInstance();
+                printMatcher(DescriptionBuilder.buildMap(builder.builtClass()), writer);
+            } catch (NoSuchMethodException | SecurityException | InstantiationException | IllegalAccessException
+                    | IllegalArgumentException | InvocationTargetException e) {
+                throw new ConfigurationException(
+                        String.format("Can not instantiate matcher builder  %s ", mClazz.getName()), e);
+            }
+        }
+
+        writer.write(String.format("%n>>>> FAMILIES (Datatype ILicenseFamily) <<<<%n%n"));
+        for (ILicenseFamily family : cfg.getLicenseFamilies(LicenseFilter.all)) {
+            printFamily(family, writer);
+        }
+    }
+
+    private static void printFamily(ILicenseFamily family, Writer writer) throws IOException {
+        writer.write(String.format("'%s' - %s%n", family.getFamilyCategory().trim(), family.getFamilyName()));
+    }
+
+    private static void printMatcher(Description matcher, Writer writer) throws IOException {
+        writer.write(String.format("'%s' - %s%n", matcher.getCommonName(), matcher.getDescription()));
+        printChildren(1, matcher.getChildren(), writer);
+
+    }
+
+    private static void printLicense(ILicense license, Writer writer) throws IOException {
+        Description dLicense = license.getDescription();
+        writer.write(String.format("'%s' - %s %n", dLicense.getCommonName(), dLicense.getDescription()));
+        printChildren(1, dLicense.getChildren(), writer);
+    }
+
+    private static void writeIndent(int indent, Writer writer) throws IOException {
+        for (int i = 0; i < indent; i++) {
+            writer.write(INDENT);
+        }
+    }
+
+    private static void printChildren(int indent, Map<String, Description> children, Writer writer) throws IOException {
+        for (Description d : children.values()) {
+            writeIndent(indent, writer);
+            switch (d.getType()) {
+            case Parameter:
+            case BuilderParam:
+                writer.write(String.format("'%s' %s (Datatype: %s%s)%n", d.getCommonName(), d.getDescription(),
+                        d.isCollection() ? "Collection of " : "", d.getChildType().getSimpleName()));
+                break;
+            case Unlabeled:
+                writer.write(String.format("'%s name' %s (Datatype: %s%s)%n", d.getCommonName(), d.getDescription(),
+                        d.isCollection() ? "Collection of " : "", d.getChildType().getSimpleName()));
+                break;
+            case License:
+                // do noting
```

Comment:
Typo ;) nothing - i though of moving it to a default block .... when reading it first.